src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -3887,6 +3890,9 @@ void C2_MacroAssembler::vector_mask_operation(int opc, Register dst, XMMRegister
   vpxor(xtmp, xtmp, xtmp, vec_enc);
   vpsubb(xtmp, xtmp, mask, vec_enc);
   vpmovmskb(tmp, xtmp, vec_enc);
+  if (masklen < 64) {
+    andq(tmp, (((jlong)1 << masklen) - 1));
+  }
```

Comment:
Hi @sviswa7 , @XiaohongGong , I think problem will occur only on non-AVX512VLBW platforms. So this workaround fix can be removed from this place.